### ISSUE TITLE
Tsconfig compilerOptions should be optional when extending another config. 

### ISSUE THREAD

**Issue by coleHafner** [projen/projen#3411]
## Problem
When creating a new typescript project `compilerOptions` is required, even when extending another tsconfig.

## Example
```typescript
export class BasePackage extends TypeScriptProject {
  constructor(parent: Parent, _options: BasePackageOptions) {
    super({
      name: "my-package-name",
      disableTsconfig: false,
      tsconfig: {
        exclude: ['node_modules'],
        extends: TypescriptConfigExtends.fromPaths(["path/to/base/config.json]),
        include: ['**/*.ts'],
      },
    });
  }
}
```

This throws an error because `compilerOptions` is not specified. And when I set it to an empty object `compilerOptions: {}`, it will inject defaults that override those in my base config. 

## Proposal
Make `compilerOptions` property optional when extending or just optional always. Is there a different way around this that I'm overlooking? Currently I can `disableTsConfig: true` and create my own custom tsconfig that omits these options using `JsonFile`:

```typescript
// create tsconfig to extend base config
const tsConfigFilename = "tsconfig.json";
const configPackageName = "@org/config-package-name"
const baseConfigFilename = "tsconfig.base.json";
const tsConfigOpts = {
   obj: {
      extends: [configPackageName, baseConfigFilename].join("/"),
      include: ["**/*.ts"],
   },
};

new JsonFile(this, tsConfigFilename, tsConfigOpts);
```

**Comment by mrgrain** [comment]
I don't see a problem with making it optional.

**Comment by coleHafner** [comment]
Okay. PR incoming...

**Comment by giseburt** [comment]
DL;DR: I would like to piggyback on this the ability (for the same reasons) to specify the **exact** `compileOptions` value I want without it merging in the defaults, in some cases with an `extends` as well.

I'm working with a similar situation.

I'm working with using the [`@tsconfig`](https://github.com/tsconfig/bases) to extend. Sadly, to get `strictest` to work we would need to use `extends` as an array, and TypeStrong/ts-node#2000 means we can't do that in a tsconfig file that's used with `ts-node`.

In a typical repo, where `ts-node` is only used by `projen` itself, that only effects the `tsconfig.dev.json`, however some repo types like the `CDK` variants use `ts-node` as well.

That said, this means we have three scenarios:

1. Use the defaults, maybe a few overrides (current state)
2. Would like to use `extends` to _set_ the `compileOptions`
    - In this case, we do **not** want to apply the default `compileOptions`
    - We can assume that `extends` can be an array of files, such as `node18` and `strictest` (ignoring the `ts-node` bug)
3. Would like to use a single `extends` to set the base `compileOptions`, and also have a `compileOptions` where we specify specific overrides from that extended config
    - Again, we don't want the default project `compileOptions` to what's provided

With the current code, I'm having issue doing this. I've had to copy the [`compilerOptionDefaults`](https://github.com/projen/projen/blob/0d9027ac26d93c30784c1e184312189bc9e31cf5/src/typescript/typescript.ts#L456-L478) in my code, replace all the values with `undefined` except the ones I want to set, just to get the output of _just_ what I provide.

**Comment by mrgrain** [comment]
Please don't piggy back => New issue and link if you think it's relevant to another issue.

**Comment by giseburt** [comment]
Let me rephrase: 

> Make compilerOptions property optional when extending or just optional always.

I would like to request / suggest “optional always.” Further, I’d like to have the optional be “no defaults” so if we _do_ provide a `compilerOptions` it optionally doesn’t have the defaults injected.

Said yet a third way: if we provide an option to disable the injection of the defaults and make `compilerOptions` optional in the type then all of the above will be covered.

@coleHafner I’ll happily lend a hand on the PR if you wish.

**Comment by mrgrain** [comment]
Please don't combine these two different requests/changes. They will need to be separate PRs with separate discussions.
